### PR TITLE
review comment of dnldd

### PR DESCRIPTION
**Pull request by huynt**
Feedback from dnldd
1. https://github.com/raedahgroup/dcrwallet/blob/dicemix/dcrtxclient/dcrtxclient.go#L31 - This looks like an initializer, for a setter you need func (c *Client) SetConfig here
2. https://github.com/raedahgroup/dcrwallet/blob/dicemix/dcrtxclient/dcrtxclient.go#L38 - Why is the client being returned here? The func already has a client receiver pointer.
3. comments should not refer to variable names, describe everything in plain english. Comments could also be more detailed.
4. there are multiple leftover commented debug lines left in the code.
5. StartSession is a duplicate of https://github.com/raedahgroup/dcrwallet/blob/dicemix/dcrtxclient/dcrtxclient.go#L38-L59
6. Why not export the config field? https://github.com/raedahgroup/dcrwallet/blob/dicemix/dcrtxclient/dcrtxclient.go#L87-L89
7. /dcrtxclient/ripemd128 should be a dependency, not includeded directly.

### CODE REVIEW

File: dcrtxclient/dcrtxclient.go [hunk 1]
```diff
@@ -30,13 +30,13 @@ type (
 
 func SetConfig(cfg *Config) *Client {
 	client := &Client{
-		cfg: cfg,
+		Cfg: cfg,
 	}
 	return client
 }
 
 func (c *Client) StartSession() (*Client, error) {
```
Comment (dnldd) [inline dcrtxclient/dcrtxclient.go]: I rewrote `StartSession` to clarify what I meant about the pointer receiver and comments.
```
// startSession establishes a connection to the transaction matching server if
// the client's configuration allows it.
```
// startSession establishes a connection to the transaction matching server if
// the client's configuration allows it.
func (c *Client) startSession() error {
	if c.Cfg.Enable {
		conn, err := c.Connect()
		if err != nil {
			return err
		}

		if conn == nil {
			return ErrCannotConnect
		}

		c.conn = conn
		err := c.registerServices()
		if err != nil {
			return err
		}
	}

	if !c.Cfg.Enable {
		fmt.Printf("Session aborted, the client is currently disabled.")
	}

	return nil
}
```

I'm not returning `c` because the function signature provides a  pointer receiver `func (c *Client) StartSession() error `, all modifications made to the the instance through the  pointer receiver persist.

File: dcrtxclient/dcrtxclient.go [hunk 2]
```diff
@@ -58,46 +58,12 @@ func (c *Client) StartSession() (*Client, error) {
 	return c, nil
 }
 
-func NewClient(cfg *Config) (*Client, error) {
```
Comment (dnldd) [inline dcrtxclient/dcrtxclient.go]: I'd unexport`StartSession` and maintain `NewClient`, like so:

```
// startSession establishes a connection to the transaction matching server if
// the client's configuration allows it.
func (c *Client) startSession() error {
	if c.Cfg.Enable {
		conn, err := c.Connect()
		if err != nil {
			return err
		}

		if conn == nil {
			return ErrCannotConnect
		}

		c.conn = conn
		err := c.registerServices()
		if err != nil {
			return err
		}
	}

	if !c.Cfg.Enable {
		fmt.Printf("Session aborted, the client is currently disabled.")
	}

	return nil
}

// NewClient initializes a dcrtxclient instance.
func NewClient(cfg *Config) (&Client, error) {
	client := Client{
		Cfg: cfg,
	}

	err := client.startSession()
	if err != nil {
		return nil, err
	}

	return client, nil
}
```

File: dcrtxclient/dcrtxclient.go [hunk 3]
```diff
@@ -30,13 +30,13 @@ type (
 
 func SetConfig(cfg *Config) *Client {
 	client := &Client{
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/dcrtxclient.go]: I understand your idea, some of my code is rollback. I'd remove this function. The initial idea of this function is setting config for client. When we export field config then this function is unnecessary.

File: dcrtxclient/dcrtxclient.go (hunk 1 above)
Comment (huyntsgs) [reply dcrtxclient/dcrtxclient.go]: You are right! Return client is unnecessary. I updated the function.

File: dcrtxclient/dcrtxclient.go (hunk 2 above)
Comment (huyntsgs) [reply dcrtxclient/dcrtxclient.go]: I have removed NewClient function because it is not using in any code. And the idea is that, only want to start connect to dcrtxmatcher at proper time, not at time creating new client. We work with dcrtxmatcher connection at time purchase ticket, and we have to set dcrtxclient config at initial of dcrwallet. This is @raedah 's suggestion.

File: dcrtxclient/dcrtxclient.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dnldd) [reply dcrtxclient/dcrtxclient.go]: Noted, let's have it removed then in favor of exporting the cfg var.

File: dcrtxclient/dcrtxclient.go (hunk 2 above)
Comment (dnldd) [reply dcrtxclient/dcrtxclient.go]: Alright that works.

File: dcrtxclient/chacharng/chacharng.go [hunk 4]
```diff
@@ -7,8 +7,8 @@ import (
 	"github.com/tmthrgd/go-rand"
 )
 
-//func returns random [rndsize]byte slice with provided seed
-//based on chacha20. The seed size must be 32.
+// func returns random [rndsize]byte slice with provided seed
+// based on chacha20. The seed size must be 32.
 func RandBytes(seed []byte, rndsize int) []byte {
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: I have added error for return parameter

File: dcrtxclient/chacharng/chacharng.go [hunk 5]
```diff
@@ -1,63 +1,62 @@
...
-	fmt.Printf("RngBytes %x\n", ret)
-
-	return r, ret
+	return r, ret, nil
 
 }
 
-//get new reader from seed.
-func NewRandReader(seed []byte) io.Reader {
+//NewRandReader gets new reader from seed.
```
Comment (dnldd) [inline dcrtxclient/chacharng/chacharng.go]: `NewRandReader` is only used by `NewReaderBytes` here, I think it should be refactored into `NewReaderBytes` as a result

Comment (dnldd) [inline dcrtxclient/chacharng/chacharng.go]: `NewRandReader creates a new rand reader from the provided seed.`

Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: updated

File: dcrtxclient/finitefield/finitefield.go [hunk 6]
```diff
@@ -35,6 +37,8 @@ func (ff Field) Sub(ff2 Field) Field {
 	return ff.Add(ff2.Neg())
 }
 
+// Perform multiplation over finite field
+// and assign result for first receiver
 func (ff *Field) MulAssign(ff2 Field) {
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/finitefield/finitefield.go]: updated and checked on other part of code

File: dcrtxclient/dcrtxclient.go [hunk 7]
```diff
@@ -21,85 +20,48 @@ type (
 
 	Client struct {
 		sync.Mutex
-		cfg  *Config
+		Cfg  *Config
 		conn *grpc.ClientConn
 		*service.TransactionService
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/dcrtxclient.go]: Updated

File: dcrtxclient/chacharng/chacharng.go [hunk 8]
```diff
@@ -1,63 +1,62 @@
...
 
 }
 
-//get new reader from seed.
-func NewRandReader(seed []byte) io.Reader {
+// NewRandReader creates a new rand reader from the provided seed.
+func NewRandReader(seed []byte) (io.Reader, error) {
 	r, err := rand.New(seed[:])
 	if err != nil {
 		fmt.Println("error new chacharng with seed")
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: Removed

File: dcrtxclient/chacharng/chacharng.go [hunk 9]
```diff
@@ -1,63 +1,62 @@
...
+		return nil, nil, err
 	}
 
 	ret := make([]byte, rndsize)
 
 	n, _ := r.Read(ret)
 
 	if n != rndsize {
-		return nil, nil
+		return nil, nil, errors.New("error returns wrong size of bytes")
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: Updated by removing error word, supplied more information.

File: dcrtxclient/chacharng/chacharng.go [hunk 10]
```diff
@@ -1,63 +1,62 @@
...
-	r := NewRandReader(seed[:])
-	if r != nil {
-		fmt.Println("error new chacharng with seed")
-		return nil, nil
+// NewReaderBytes generates random [rndsize]byte slice with provided seed and reader for next random
+// based on chacha20. The seed size must be 32.
+func NewReaderBytes(seed []byte, rndsize int) (io.Reader, []byte, error) {
+	r, err := NewRandReader(seed[:])
+	if err != nil {
+		fmt.Println("error creates chacharng bytes with seed")
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: Remove

File: dcrtxclient/chacharng/chacharng.go [hunk 11]
```diff
@@ -1,63 +1,62 @@
...
-//NewReaderBytes generates random [rndsize]byte slice with provided seed and reader for next random
-//based on chacha20. The seed size must be 32.
-func NewReaderBytes(seed []byte, rndsize int) (io.Reader, []byte) {
-	r := NewRandReader(seed[:])
-	if r != nil {
-		fmt.Println("error new chacharng with seed")
-		return nil, nil
+// NewReaderBytes generates random [rndsize]byte slice with provided seed and reader for next random
+// based on chacha20. The seed size must be 32.
+func NewReaderBytes(seed []byte, rndsize int) (io.Reader, []byte, error) {
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: Updated

File: dcrtxclient/chacharng/chacharng.go [hunk 12]
```diff
@@ -1,63 +1,62 @@
...
+		return nil, err
 	}
 
 	ret := make([]byte, rndsize)
 
 	n, err := r.Read(ret)
 
 	if n != rndsize {
-		return nil
+		return nil, errors.New("error returns wrong size of bytes")
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: Updated as comment

File: dcrtxclient/chacharng/chacharng.go [hunk 13]
```diff
@@ -1,63 +1,62 @@
...
-//based on chacha20. The seed size must be 32.
-func RandBytes(seed []byte, rndsize int) []byte {
+// RandBytes returns random [rndsize]byte slice with provided seed
+// based on chacha20. The seed size must be 32.
+func RandBytes(seed []byte, rndsize int) ([]byte, error) {
 	r, err := rand.New(seed[:])
 	if err != nil {
-		fmt.Println("error new chacharng with seed")
-		return nil
+		fmt.Println("error creates chacharng bytes with seed")
```
[the remark this answers is not in the document]
Comment (huyntsgs) [reply dcrtxclient/chacharng/chacharng.go]: Removed
